src/chunk_constraint.c

```diff
@@ -782,7 +803,8 @@ ts_chunk_constraint_delete_by_dimension_slice_id(int32 dimension_slice_id)
 		ts_scan_iterator_create(CHUNK_CONSTRAINT, RowExclusiveLock, CurrentMemoryContext);
 	int count = 0;
 
-	init_scan_by_dimension_slice_id(&iterator, dimension_slice_id);
+	ts_chunk_constraint_scan_iterator_set_slice_id(&iterator, dimension_slice_id);
+
```

Comment:
Codecov complains here, does this mean this function is never used? May be a good time to delete it while we've noticed.

Reply:
Yeah, I think that might be the case. Probably related to the change a while back where chunks are tombstoned (with the dropped=true) instead of deleted. 

I'd prefer to leave further refactoring out of this PR, however.